### PR TITLE
Basic auth support for ES endpoints

### LINKED ISSUE #10: unable to use this plugin with elasticsearch shield security
does not support giving basic authentication as username:pass@localhost in config file. 

a work around for this is to configure anonymous authentication for shield, however I don't know what permissions to create in the roles as I'm not sure which operations are being performed on startup, which indices and cluster operations must I allow? 

error on running with a shield secured node in foreground (esearch 1.5)

Exception in thread "main" java.lang.IllegalArgumentException: URI can't be null.
        at sun.net.spi.DefaultProxySelector.select(DefaultProxySelector.java:147)
        at sun.net.www.protocol.http.HttpURLConnection.plainConnect0(HttpURLConnection.java:1098)
        at sun.net.www.protocol.http.HttpURLConnection.plainConnect(HttpURLConnection.java:998)
        at sun.net.www.protocol.http.HttpURLConnection.connect(HttpURLConnection.java:932)
        at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1512)
        at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1440)
        at me.snov.newrelic.elasticsearch.parsers.AbstractParser.getInputStream(Unknown Source)
        at me.snov.newrelic.elasticsearch.parsers.AbstractParser.request(Unknown Source)
        at me.snov.newrelic.elasticsearch.ElasticsearchAgent.getClusterName(Unknown Source)
        at me.snov.newrelic.elasticsearch.ElasticsearchAgent.<init>(Unknown Source)
        at me.snov.newrelic.elasticsearch.ElasticsearchAgentFactory.createConfiguredAgent(Unknown Source)
        at com.newrelic.metrics.publish.AgentFactory.createAndRegister(AgentFactory.java:97)
        at com.newrelic.metrics.publish.AgentFactory.createConfiguredAgents(AgentFactory.java:57)
        at com.newrelic.metrics.publish.Runner.createAgents(Runner.java:135)
        at com.newrelic.metrics.publish.Runner.setupAgents(Runner.java:142)
        at com.newrelic.metrics.publish.Runner.setupAndRun(Runner.java:103)
        at me.snov.newrelic.elasticsearch.Main.main(Unknown Source)
[13:27:29] [ERROR] npi - [Error: Error while starting plugin process in foreground]

### PR DESCRIPTION
**Pull request by preflightsiren**
simple addition to support basic authentication headers in the requests to elastic search.

resolves #10

### CODE REVIEW

File: build.xml [hunk 1]
```diff
@@ -113,7 +117,8 @@
             </manifest>
             <fileset dir="${main.build.classes.dir}"/>
             <zipfileset src="${sdk.jar}" />
-            <zipfileset src="${gson.jar}" />
+                <zipfileset src="${gson.jar}" />
```
Comment (s12v) [inline build.xml]: extra indent

**Comment by s12v** [comment]
@preflightsiren, thanks for the PR! Would it make sense to put empty username/password to the configuration template?

**Comment by preflightsiren** [comment]
thanks for the quick review @s12v ! I've incorporated your suggestions (fixed some whitespace, added examples to the config template).

One improvement that could be made is to add to the JUnit tests to ensure that the authorization headers are only added when username and password is defined. I'm not super confident with JUnit so any help would be appreciated :D

**Comment by preflightsiren** [comment]
@s12v was there anything else you wanted to see for this PR?

**Comment by s12v** [comment]
@preflightsiren, no, thank you. I will test/merge it next week, I had very limited internet coverage this week :)

**Comment by s12v** [comment]
@preflightsiren, thank you

**Comment by jdavisclark** [comment]
@s12v: when will this be published to Plugin Central?

**Pull request merged**